h2o-core/src/main/java/water/rapids/ast/prims/reducers/AstCumu.java

```diff
@@ -44,8 +44,8 @@ public ValFrame apply(Env env, Env.StackHelp stk, AstRoot asts[]) {
       if(v.isCategorical() || v.isString() || v.isUUID()) throw new IllegalArgumentException(
               "Cumulative functions not applicable to enum, string, or UUID values");
     }
+    if (axisAR.exec(env).getNum() != 1 && axisAR.exec(env).getNum() != 0) throw new IllegalArgumentException("Axis must be 0 or 1");
```

Comment:
No, I think the previous version is better.
In general it is not safe to evaluate the expression more than once: it may not be an AstNum but some complicated expression returning a ValNum. And that complicated expression may have side-effects...